src/frontend/components/UI/LogFileUploadDialog/index.tsx

```diff
@@ -0,0 +1,109 @@
+import { useTranslation } from 'react-i18next'
+import { CircularProgress } from '@mui/material'
+
+import { Dialog, DialogContent, DialogFooter } from '../Dialog'
+import { useShallowGlobalState } from 'frontend/state/GlobalStateV2'
+
+export default function LogUploadDialog() {
+  const { t } = useTranslation()
+  const { uploadLogFileProps, setUploadLogFileProps } = useShallowGlobalState(
+    'uploadLogFileProps',
+    'setUploadLogFileProps'
+  )
+
+  const [confirmed, setConfirmed] = useState(false)
+  const [uploading, setUploading] = useState(false)
+  const [error, setError] = useState(false)
+  const [uploadUrl, setUploadUrl] = useState<string | null>(null)
+
+  const onClose = useCallback(() => {
+    setConfirmed(false)
+    setUploading(false)
+    setUploadLogFileProps(false)
+  }, [setConfirmed, setUploading, setUploadLogFileProps])
+
+  const doUpload = useCallback(async () => {
+    if (!uploadLogFileProps) return
+
+    setConfirmed(true)
+    setUploading(true)
+
+    const result = await window.api.uploadLogFile(
+      uploadLogFileProps.name,
+      uploadLogFileProps.appNameOrRunner
+    )
+
+    setUploading(false)
+    if (!result) {
+      setError(true)
+      return
+    }
+    const [url] = result
+    await navigator.clipboard.writeText(url)
+    setUploadUrl(url)
+  }, [uploadLogFileProps])
+
+  const [dialogContent, dialogFooter] = useMemo(() => {
+    if (!uploadLogFileProps) return ['', '']
+    if (!confirmed)
+      return [
+        t(
+          'setting.log.upload.confirm',
+          'Do you really want to upload "{{name}}"?',
+          { name: uploadLogFileProps.name }
+        ),
+        <>
+          <button onClick={doUpload} className={'button is-primary'}>
+            {t('box.yes')}
+          </button>
+          <button onClick={onClose} className={'button is-danger'}>
+            {t('box.no')}
+          </button>
+        </>
+      ]
+    if (uploading)
+      return [
+        <>
+          <CircularProgress />
+          {t('setting.log.upload.uploading', 'Uploading log file...')}
+        </>,
+        <></>
+      ]
+    if (error)
+      return [
+        t(
+          'setting.log.upload.error',
+          "Failed to upload log file. Check Heroic's general log for details"
+        ),
+        <>
+          <button onClick={onClose} className={'button is-secondary'}>
+            {t('box.ok')}
+          </button>
+        </>
+      ]
+    return [
+      t(
+        'setting.log.upload.done',
+        'Upload complete. Uploaded to {{url}} (URL copied to your clipboard)',
+        {
+          url: uploadUrl
+        }
+      ),
+      <>
+        <button onClick={onClose} className={'button is-secondary'}>
+          {t('box.ok')}
+        </button>
+      </>
+    ]
+  }, [uploadLogFileProps, confirmed, uploading, error, uploadUrl])
+
+  if (!uploadLogFileProps) return <></>
+
+  return (
+    <Dialog onClose={onClose} showCloseButton={false}>
```

Comment:
we can maybe add a title/header here, the dialog looks a bit weird

![image](https://github.com/user-attachments/assets/92e58605-037c-44d1-a880-14525bc7d8fc)

the `Upload completed` sentence could be the title for example

not a blocker though, could be improved later